packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -122,6 +123,7 @@ function attributeExpressionReferencesForEachIndex(
 export default function parse(source: string, state: State): TemplateParseResult {
     const warnings: CompilerDiagnostic[] = [];
     const seenIds: Set<string> = new Set();
+    const removedAttrs: Set<string> = new Set();
```

Comment:
At this end of this refactor I would like to get rid of this closure.  So instead of keeping track of the removed attributes, I would recommend copying the attribute list and keeping removing the attributes from the copied list.

Also, something worth pointing out is that `getTemplateAttribute` transforms parse5 attribute nodes into `IRAttribute` on the fly. A more elegant approach would be to map all the parse5 attributes to `IRAttribute` at the beginning of the `parseElement` method.

Reply:
Updated to remove the set and now generating the IRAttributes at the beginning of the parsing.